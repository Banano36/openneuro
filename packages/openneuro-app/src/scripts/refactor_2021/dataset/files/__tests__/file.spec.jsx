import React from 'react'
import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import File from '../file.jsx'

describe('File component', () => {
  it('renders with common props', () => {
    const { asFragment } = render(
      <File datasetId="ds001" path="" filename="README" />,
      { wrapper: MemoryRouter },
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it('renders for dataset snapshots', () => {
    const { asFragment } = render(
      <File datasetId="ds001" snapshotTag="1.0.0" path="" filename="README" />,
      { wrapper: MemoryRouter },
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it('generates correct download links for top level files', () => {
    render(<File datasetId="ds001" path="" filename="README" />, {
      wrapper: MemoryRouter,
    })
    expect(screen.getByRole('link', { name: 'download file' })).toHaveAttribute(
      'href',
      '/crn/datasets/ds001/files/README',
    )
  })
  it('generates correct download links for nested files', () => {
    render(
      <File
        datasetId="ds001"
        path="sub-01:anat"
        filename="sub-01_T1w.nii.gz"
      />,
      { wrapper: MemoryRouter },
    )
    expect(screen.getByRole('link', { name: 'download file' })).toHaveAttribute(
      'href',
      '/crn/datasets/ds001/files/sub-01:anat:sub-01_T1w.nii.gz',
    )
  })
})
