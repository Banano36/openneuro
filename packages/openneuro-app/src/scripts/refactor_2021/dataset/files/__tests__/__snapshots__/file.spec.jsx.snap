// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File component renders for dataset snapshots 1`] = `
<Fragment>
  <Icon
    color="deepskyblue"
    icon="fab fa-markdown"
  />
    
  README
  <span
    className="filetree-editfile"
  >
    <Media
      className=""
      greaterThanOrEqual="medium"
    >
      <Tooltip
        tooltip="Download"
      >
        <span
          className="edit-file download-file"
        >
          <a
            download={true}
            href="/crn/datasets/ds001/snapshots/1.0.0/files/README"
          >
            <i
              className="fa fa-download"
            />
          </a>
        </span>
      </Tooltip>
    </Media>
    <Tooltip
      tooltip="View"
    >
      <span
        className="edit-file view-file"
      >
        <Link
          to="/datasets/ds001/versions/1.0.0/file-display/README"
        >
          <i
            className="fa fa-eye"
          />
        </Link>
      </span>
    </Tooltip>
  </span>
</Fragment>
`;

exports[`File component renders with common props 1`] = `
<Fragment>
  <Icon
    color="deepskyblue"
    icon="fab fa-markdown"
  />
    
  README
  <span
    className="filetree-editfile"
  >
    <Media
      className=""
      greaterThanOrEqual="medium"
    >
      <Tooltip
        tooltip="Download"
      >
        <span
          className="edit-file download-file"
        >
          <a
            download={true}
            href="/crn/datasets/ds001/files/README"
          >
            <i
              className="fa fa-download"
            />
          </a>
        </span>
      </Tooltip>
    </Media>
    <Tooltip
      tooltip="View"
    >
      <span
        className="edit-file view-file"
      >
        <Link
          to="/datasets/ds001/file-display/README"
        >
          <i
            className="fa fa-eye"
          />
        </Link>
      </span>
    </Tooltip>
  </span>
</Fragment>
`;
